tracer/test/Datadog.Trace.Security.Unit.Tests/IAST/Tainted/DefaultTaintedMapTests.cs

```diff
@@ -0,0 +1,337 @@
+// <copyright file="DefaultTaintedMapTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Datadog.Trace.Iast;
+using Xunit;
+using Range = Datadog.Trace.Iast.Range;
+
+namespace Datadog.Trace.Security.Unit.Tests.Iast.Tainted;
+
+public class DefaultTaintedMapTests
+{
+    [Fact]
+    public void GivenATaintedObjectMap_WhenPutAndGetString_ObjectIsRetrieved()
+    {
+        DefaultTaintedMap map = new();
+        string testString = "test";
+        var source = new Source(12, "name", "value");
+        var tainted = new TaintedObject(testString, new Range[] { new Range(1, 2, source) });
+        map.Put(tainted);
+        var tainted2 = map.Get(testString);
+        Assert.NotNull(tainted2);
+        Assert.Equal(testString, tainted2.Value);
```

Comment:
nit: we generally favour fluent assertions (e.g. for other tests ideally)
```suggestion
        tainted2.Should().NotBeNull();
        tainted2.Value.Should().Be(testString);
```